### DISCUSSION TITLE
Tooltips for mobile

### DISCUSSION THREAD

**Discussion by whiztech** [hbstack/.github#37] [Ideas]
I'm interested on having mobile tooltips for the icons in [revision module](https://github.com/hbstack/revision) in case a visitor can't guess what the icons are representing. This is useful for [Abbreviations Shortcode](https://hugomods.com/docs/shortcodes/abbr/) too.

What would be the best way to do this? Because this module is based on Bootstrap, I thought of just using [Bootstrap's tooltips](https://getbootstrap.com/docs/5.3/components/tooltips/):

![image](https://github.com/hbstack/.github/assets/159569/c00ad38a-9cb9-4981-bcf6-547b224657f1)

![image](https://github.com/hbstack/.github/assets/159569/585d1835-729b-4ce5-8f38-96c1241315fa)

Thank you.

**Comment by razonyang** [comment]
I don't have an ideal solution yet, since it will import new JS, I don't think everyone like/need it.

Using JS API to dynamic generate tooltips elements would be a workaround, something like.

```typescript
import Tooltip from "js/bootstrap/src/tooltip"

const elements = Array.from(document.querySelectorAll('[data-bs-toggle="tooltip"]'))

elements.push(...document.querySelectorAll('abbr'))

elements.forEach((ele) => {
  new Tooltip(ele)
})
```

**Comment by razonyang** [comment]
I've an idea few weeks ago, just got time to do it, [github.com/hbstack/revision@v0.4.0](https://github.com/hbstack/revision/releases/tag/v0.4.0) now automatically using tooltip if `github.com/hbstack/bs-tooltip` module was deteced, that mean, you'll need to import the tooltip module to tooltip revision titles. It won't affect users whose don't want to use Bootstrap tooltip.

**Comment by razonyang** [comment]
To use Bootstrap tooltip on `<abbr>`, you can import the `github.com/hbstack/bs-tooltip/modules/abbr` module.
